test/integration/TestHeap.t.sol

```diff
@@ -5,314 +5,281 @@ import {Test} from "forge-std/Test.sol";
         }
+        heap.increase(ids[0], type(uint256).max);
 
-        assertEq(heap.getRoot(), accounts[0]);
+        assertEq(heap.length(), 3);
+        assertEq(heap.getRoot(), ids[0]);
+        assertEq(heap.getParent(ids[1]), ids[0]);
+        assertEq(heap.getParent(ids[2]), ids[0]);
+        assertTrue(_compareUnorderedTuples(heap.getLeftChild(ids[0]), heap.getRightChild(ids[0]), ids[1], ids[2]));
+    }
 
-        for (uint256 i = 0; i < accounts.length; i++) {
-            assertEq(heap.accounts[i].id, accounts[i]);
+    function testInsertMany(uint256[TESTED_SIZE] calldata values) public {
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            heap.insert(ids[i], values[i]);
+            _assertNodeCorrect(ids[i], values[i]);
         }
     }
 
-    function testShouldRemoveAllSortedAccount() public {
-        for (uint256 i = 0; i < accounts.length; i++) {
-            heap.insert(accounts[i], TESTED_SIZE - i);
+    function testRemoveMany(uint256[TESTED_SIZE] calldata values) public {
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            heap.insert(ids[i], values[i]);
         }
 
-        for (uint256 i = 0; i < accounts.length; i++) {
-            heap.remove(accounts[i]);
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            address id = ids[i];
+            heap.remove(id);
+            assertFalse(heap.containsAccount(id));
+            assertEq(heap.getParent(id), address(0));
+            assertEq(heap.getLeftChild(id), address(0));
+            assertEq(heap.getRightChild(id), address(0));
         }
-
-        assertEq(heap.length(), 0);
-        assertEq(heap.getRoot(), ADDR_ZERO);
     }
 
-    function testShouldInsertAccountSortedAtTheBeginning() public {
-        uint256 value = 50;
-
-        // Add first 10 accounts with decreasing value.
-        for (uint256 i = 0; i < 10; i++) {
-            heap.insert(accounts[i], value - i);
+    function testIncreaseMany(uint256[TESTED_SIZE] memory initialValues, uint256[TESTED_SIZE] memory increasedValues)
+        public
+    {
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            initialValues[i] = bound(initialValues[i], 0, type(uint256).max - 1);
+            increasedValues[i] = bound(increasedValues[i], initialValues[i] + 1, type(uint256).max);
         }
 
-        // Add last 10 accounts at the same value.
-        for (uint256 i = TESTED_SIZE - 10; i < TESTED_SIZE; i++) {
-            heap.insert(accounts[i], 10);
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            heap.insert(ids[i], initialValues[i]);
         }
 
-        assertEq(heap.getRoot(), accounts[0], "root not expected");
-
-        for (uint256 i = 0; i < 10; i++) {
-            assertEq(heap.accounts[i].id, accounts[i], "order not expected, 1");
+        for (uint256 i; i < TESTED_SIZE; ++i) {
+            heap.increase(ids[i], increasedValues[i]);
+            _assertNodeCorrect(ids[i], increasedValues[i]);
```

Comment:
(optional) We could also do this check in another loop, so that we check the entire structure and do not do modifications to the tree in between the checks. This remark also applies to `testDecreaseMany`

Reply:
I think it would be too resource-intensive, so I won't implement this one